packages/angular_devkit/benchmark/bin/benchmark.ts

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env node
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { logging, tags, terminal } from '@angular-devkit/core';
+import { appendFileSync, writeFileSync } from 'fs';
+import * as minimist from 'minimist';
+import { filter, map } from 'rxjs/operators';
+import { Command } from '../src/command';
+import { defaultReporter } from '../src/default-reporter';
+import { defaultStatsCapture } from '../src/default-stats-capture';
+import { runBenchmark } from '../src/run-benchmark';
+
+// Show usage of the CLI tool, and exit the process.
+function usage(logger: logging.Logger, exitCode = 0): never {
+  logger.info(tags.stripIndent`
+    benchmark [options] -- [command to benchmark]
+
+    Collects process stats from running the command.
+
+    Options:
+        --help                    Show this message.
+        --verbose                 Show more information while running.
+        --exit-code               Expected exit code for the command. Default is 0.
+        --iterations              Number of iterations to run the benchmark over. Default is 5.
+        --retries                 Number of times to retry when process fails. Default is 5.
+        --cwd                     Current working directory to run the process in.
+        --output-file             File to output benchmark log to.
+        --overwrite-output-file   If the output file should be overwritten rather than appended to.
+        --prefix                  Logging prefix.
+
+    Example:
+        benchmark --iterations=3 -- node my-script.js
+  `);
+
+  process.exit(exitCode);
```

Comment:
I want tests and users to be able to call the main. That lets me call the program the same way it is from the command-line, not missing any bits like arg handling when I call it via API call.

Could you end the script with 
```
if (require.main === module) {
  process.exitCode = main(process.argv.slice(2));
}
```
and remove all the process.exit() calls? Those make it impossible to re-use this in another context

Also this demonstrates that there is no integration test that calls the top-level of the program

Reply:
Wrapped binary code into a exported main function, calling it only if this is the main for the node program.